src/fastfield/readers.rs

```diff
@@ -46,18 +53,70 @@ impl FastFieldReaders {
         Ok(PerFieldSpaceUsage::new(per_field_usages))
     }
 
+    pub(crate) fn columnar(&self) -> &ColumnarReader {
+        self.columnar.as_ref()
+    }
+
+    /// Transforms a user-supplied fast field name into a column name.
+    ///
+    /// A user-supplied fast field name is not necessarily a schema field name
+    /// because we handle fast fields.
+    ///
+    /// For instance, if the documents look like `{.., "attributes": {"color": "red"}}` and
+    /// `attributes` is a json fast field,  a user could want to run a term aggregation over
+    /// colors, by referring to the field as `attributes.color`.
+    ///
+    /// This function transforms `attributes.color` into a column key to be used in the `columnar`.
+    ///
+    /// The logic works as follows, first we identify which field is targetted by calling
+    /// `schema.find_field(..)`. This method will attempt to split the user splied fast field
+    /// name by non-escaped dots, and find the longest matching schema field name.
+    /// In our case, it would return the (attribute_field, "color").
+    ///
+    /// If no field is found, but a dynamic field is supplied, then we
+    /// will simply assuem the user is targetting the dynamic field. (This feature is used in
+    /// Quickwit.)
+    ///
+    /// We then encode the `(field, path)` into the right `columnar_key`.
+    fn resolve_column_name_given_default_field<'a>(
```

Comment:
shouldn't this be part of `Schema` (maybe replace find_field)?

Reply:
Not really no: that logic is very specific to the quirky way we encode column names.

`schema` does not need to "know" about that.